### PR TITLE
refactor!: Rename Lend to Supply

### PR DESCRIPTION
**Pull request by toteki**
Lots of renaming here - the word "lend" no longer appears except in old changelogs and the phrase "lending pool".

The major API breaks:
- `MsgLendAsset` -> `MsgSupply`
- `QueryLoaned` -> `QuerySupplied` (and total loaned, other queries)
- `LendAPY` -> `SupplyAPY`
- Field name `Lender` -> `Supplier` field in some places

If looking for API breaking changes, view the proto file diffs first.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/architecture/ADR-009-liquidity-mining.md [hunk 1]
```diff
@@ -92,22 +92,22 @@ Amounts are `uToken` balances, exact integers which will not experience rounding
...
-- Verify lender has sufficient locked uTokens of the selected tier that are not currently unlocking
-- Start an unlocking for the lender in question
+- Verify user has sufficient locked uTokens of the selected tier that are not currently unlocking
+- Start an unlocking for the user in question
 
 Unlockings are defined as a struct:
 ```go
 type CollateralUnlocking struct {
-  Lender sdk.AccAddress
+  Supplier sdk.AccAddress
```
Comment (toteki) [inline docs/architecture/ADR-009-liquidity-mining.md:110]:
```suggestion
  User sdk.AccAddress
```

File: docs/architecture/ADR-009-liquidity-mining.md [hunk 2]
```diff
@@ -10,13 +10,13 @@ Proposed
 
 ## Abstract
 
-Umee wishes to add support for liquidity mining incentives; i.e. additional rewards on top of the normal `x/leverage` lending APY for supplying base assets.
+Umee wishes to add support for liquidity mining incentives; i.e. additional rewards on top of the normal `x/leverage` supplying APY for supplying base assets.
```
Comment (toteki) [inline docs/architecture/ADR-009-liquidity-mining.md:13]:
```suggestion
Umee wishes to add support for liquidity mining incentives; i.e. additional rewards on top of the normal `x/leverage` APY for supplying base assets.
```

File: proto/umee/leverage/v1/leverage.proto [hunk 3]
```diff
@@ -135,7 +135,7 @@ message Token {
   // This should only be used to eliminate an asset completely. A blacklisted
   // asset is treated as though its oracle price is zero, and thus ignored by
   // calculations such as collateral value and borrow limit. Can still be repaid
-  // or withdrawn, but not liquidated. A blacklisted token must have enable_lend
+  // or withdrawn, but not liquidated. A blacklisted token must have enable_supply
   // and enable_borrow set to false. Such tokens can be safely removed from the
```
Comment (toteki) [inline proto/umee/leverage/v1/leverage.proto:139]:
```suggestion
  // or withdrawn, but not liquidated. A blacklisted token must have enable_msg_supply
  // and enable_msg_borrow set to false. Such tokens can be safely removed from the
```

File: proto/umee/leverage/v1/tx.proto [hunk 4]
```diff
@@ -35,39 +35,39 @@ service Msg {
...
-  // Lender is the account address supplying assets and the signer of the message.
-  string                   lender = 1;
+message MsgSupply {
+  // Supplier is the account address supplying assets and the signer of the message.
+  string                   supplier = 1;
   cosmos.base.v1beta1.Coin amount = 2 [(gogoproto.nullable) = false];
 }
 
-// MsgWithdrawAsset represents a lender's request to withdraw lent assets.
+// MsgWithdrawAsset represents a user's request to withdraw lent assets.
```
Comment (toteki) [inline proto/umee/leverage/v1/tx.proto:46]:
```suggestion
// MsgWithdrawAsset represents a user's request to withdraw supplied assets.
```

File: proto/umee/leverage/v1/query.proto [hunk 5]
```diff
@@ -22,32 +22,32 @@ service Query {
...
   // all borrowed tokens is returned.
   rpc BorrowedValue(QueryBorrowedValueRequest)
       returns (QueryBorrowedValueResponse) {
     option (google.api.http).get = "/umee/leverage/v1/borrowed_value";
   }
 
-  // Loaned queries for the amount of tokens loaned by a user by denomination.
-  // If the denomination is not supplied, the total for each loaned token is
+  // Supplied queries for the amount of tokens  by a user by denomination.
+  // If the denomination is not specified, the total for each  token is
```
Comment (toteki) [inline proto/umee/leverage/v1/query.proto:40]:
```suggestion
  // If the denomination is not specified, the total for each supplied token is
```

**Review by toteki — COMMENTED** [review]
Massive diff - just finished combing through the files manually.

The other proto rename PRs won't be like this. This one requires a change in vocabulary in the docs.

File: docs/architecture/ADR-003-borrow-assets.md [hunk 6]
```diff
@@ -65,14 +65,14 @@ collateral_utilization(tokenA) = total_collateral(tokenA) / available_supply(tok
...
-Collateral utilization of tokenA is growing when lenders withdraw their tokenA collateral or when borrowers take a new loan of tokenA.
+Collateral utilization of tokenA is growing when suppliers withdraw their tokenA collateral or when borrowers take a new loan of tokenA.
 If a `tokenA` is not used a collateral then it's _collateral utilization_ is zero.
 It is bigger than 1 when available supply is lower than the amount of `tokenA` used as a collateral.
 When it is `N`, it means that only `1/N` of the collateral is available for redemption (u/tokenA -> tokenA).
 
 #### Examples
 
-Let's say we have 1000A (token A) provided to a system for lending. Below let's consider a state with total amount of A borrowed (B) and total amount of B used as a collateral (C) and computed collateral utilization (CU):
+Let's say we have 1000A (token A) supplied to the system. Below let's consider a state with total amount of A borrowed (B) and total amount of B used as a collateral (C) and computed collateral utilization (CU):
```
Comment (robert-zaremba) [inline docs/architecture/ADR-003-borrow-assets.md:75]:
```suggestion
Let's say we have 1000A (token A) supplied to the system (for lending or collateral). Below let's consider a state with total amount of A borrowed (B) and total amount of B used as a collateral (C) and computed collateral utilization (CU):
```

File: proto/umee/leverage/v1/leverage.proto [hunk 7]
```diff
@@ -120,10 +120,10 @@ message Token {
   string symbol_denom = 10 [(gogoproto.moretags) = "yaml:\"symbol_denom\""];
   uint32 exponent     = 11 [(gogoproto.moretags) = "yaml:\"exponent\""];
 
-  // Allows lending and setting a collateral using this token. Note that
-  // withdrawing is always enabled. Disabling lending would be one step in
+  // Allows supplying and setting a collateral using this token. Note that
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/leverage.proto:123]:
```suggestion
  // Allows supplying for lending or collateral using this token. Note that
```

File: proto/umee/leverage/v1/tx.proto [hunk 8]
```diff
@@ -8,10 +8,10 @@ option go_package = "github.com/umee-network/umee/v2/x/leverage/types";
 
 // Msg defines the x/leverage module's Msg service.
 service Msg {
-  // LendAsset defines a method for lending coins to the capital facility.
-  rpc LendAsset(MsgLendAsset) returns (MsgLendAssetResponse);
+  // Supply defines a method for supplying coins to the capital facility.
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/tx.proto:11]:
```suggestion
  // Supply moves tokens from user balance to the module balance for lending or collateral.
```

File: proto/umee/leverage/v1/tx.proto [hunk 9]
```diff
@@ -35,39 +35,39 @@ service Msg {
   rpc Liquidate(MsgLiquidate) returns (MsgLiquidateResponse);
 }
 
-// MsgLendAsset represents a lender's request to lend a base asset type to the
+// MsgSupply represents a user's request to supply a base asset type to the
 // module.
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/tx.proto:39]:
```suggestion
// MsgSupply is the request strucutre for the MsgLendAsset RPC.
```

Comment (robert-zaremba) [reply proto/umee/leverage/v1/tx.proto:39]: I think we can use it as a simpler form in other places as well. The RPC doc should be defined in the RPC method, not in the struct.

**Review by robert-zaremba — APPROVED** [review]
Approving, but left some comments and suggestions (could be applied to other places).
Moreover in some place we should add more context because supply is very general (eg: "for lending or collateral")

**Pull request merged**